libbpf-tools/biotop.c

```diff
@@ -0,0 +1,419 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+
+/*
+ * biotop Trace block I/O by process.
+ * Copyright (c) 2022 Francis Laniel <flaniel@linux.microsoft.com>
+ *
+ * Based on biotop(8) from BCC by Brendan Gregg.
+ * 03-Mar-2022   Francis Laniel   Created this.
+ */
+#include <argp.h>
+#include <errno.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+#include "biotop.h"
+#include "biotop.skel.h"
+#include "trace_helpers.h"
+
+#define warn(...) fprintf(stderr, __VA_ARGS__)
+#define OUTPUT_ROWS_LIMIT 10240
+
+enum SORT {
+	ALL,
+	IO,
+	BYTES,
+	TIME,
+};
+
+struct disk {
+	int major;
+	int minor;
+	char name[256];
+};
+
+struct disk **disks = NULL;
```

Comment:
Could this be statically allocated ? Like:
```
struct disk disks[1024];
```

So that we can avoid realloc and free below.